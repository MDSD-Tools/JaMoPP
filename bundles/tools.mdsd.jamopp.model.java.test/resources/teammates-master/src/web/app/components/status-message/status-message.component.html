<div class="alert alert-{{ statusMessage.color }}" *ngFor="let statusMessage of messages">
  <button class="close" (click)="dismiss(statusMessage)">
    <span aria-hidden="true">&times;</span>
  </button>
  <div style="padding-right: 30px;" [innerHtml]="statusMessage.message"></div>
</div>
