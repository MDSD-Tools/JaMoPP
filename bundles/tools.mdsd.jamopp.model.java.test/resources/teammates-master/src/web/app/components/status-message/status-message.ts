/**
 * Represents a status message and a severity color.
 *
 * The color is adapted from Bootstrap's color scheme.
 */
export interface StatusMessage {
  message: string;
  color: string;
}
