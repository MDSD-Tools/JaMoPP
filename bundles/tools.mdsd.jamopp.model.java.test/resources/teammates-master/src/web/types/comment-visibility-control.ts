/**
 * The comment visibility controls.
 */
export enum CommentVisibilityControl {
  /**
   * Show comment visibility control.
   */
  SHOW_COMMENT = 'SHOW_COMMENT',

  /**
   * Show giver name visibility control.
   */
  SHOW_GIVER_NAME = 'SHOW_GIVER_NAME',
}
