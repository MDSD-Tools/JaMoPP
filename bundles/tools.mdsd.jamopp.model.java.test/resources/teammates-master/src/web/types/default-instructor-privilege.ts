import { DefaultInstructorPermissions } from './api-const';
import { InstructorPermissionSet } from './api-output';

/**
 * Structure for default co-owner privileges.
 */
export const DEFAULT_PRIVILEGE_COOWNER: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.COOWNER);
    };

/**
 * Structure for default manager privileges.
 */
export const DEFAULT_PRIVILEGE_MANAGER: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.MANAGER);
    };

/**
 * Structure for default observer privileges.
 */
export const DEFAULT_PRIVILEGE_OBSERVER: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.OBSERVER);
    };

/**
 * Structure for default tutor privileges.
 */
export const DEFAULT_PRIVILEGE_TUTOR: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.TUTOR);
    };

/**
 * The default instructor privilege.
 */
export const DEFAULT_INSTRUCTOR_PRIVILEGE: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.CUSTOM);
    };
