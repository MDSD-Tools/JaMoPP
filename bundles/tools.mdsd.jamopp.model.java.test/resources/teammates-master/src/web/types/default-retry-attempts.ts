/**
 * Default maximum number of times data loading can be retried
 */
export const DEFAULT_NUMBER_OF_RETRY_ATTEMPTS: number = 3;
