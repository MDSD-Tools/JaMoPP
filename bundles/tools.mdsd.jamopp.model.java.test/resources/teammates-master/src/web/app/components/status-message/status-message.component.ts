import { Component, Input } from '@angular/core';
import { StatusMessage } from './status-message';

/**
 * List of status messages.
 */
@Component({
  selector: 'tm-status-message',
  templateUrl: './status-message.component.html',
  styleUrls: ['./status-message.component.scss'],
})
export class StatusMessageComponent {

  @Input() messages: StatusMessage[] = [];

  constructor() { }

  /**
   * Dismisses the status message.
   */
  dismiss(message: StatusMessage): void {
    this.messages.splice(this.messages.indexOf(message), 1);
  }

}
