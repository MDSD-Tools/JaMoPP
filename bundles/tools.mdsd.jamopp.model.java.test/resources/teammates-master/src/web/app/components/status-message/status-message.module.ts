import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { StatusMessageComponent } from './status-message.component';

/**
 * Module for status messages.
 */
@NgModule({
  imports: [
    CommonModule,
  ],
  exports: [
    StatusMessageComponent,
  ],
  declarations: [
    StatusMessageComponent,
  ],
})
export class StatusMessageModule { }
