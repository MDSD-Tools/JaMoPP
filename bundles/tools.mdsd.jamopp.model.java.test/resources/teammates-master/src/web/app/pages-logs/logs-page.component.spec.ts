import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { of } from 'rxjs';
import { LogService } from '../../services/log.service';
import { StatusMessageService } from '../../services/status-message.service';
import { TimezoneService } from '../../services/timezone.service';
import { GeneralLogEntry, LogEvent, LogSeverity } from '../../types/api-output';
import { LogsPageComponent } from './logs-page.component';
import { LogsPageModule } from './logs-page.module';
import Spy = jasmine.Spy;

describe('LogsPageComponent', () => {
  let component: LogsPageComponent;
  let fixture: ComponentFixture<LogsPageComponent>;
  let logService: LogService;
  let timezoneService: TimezoneService;
  let statusMessageService: StatusMessageService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [NgbModule, LogsPageModule, HttpClientTestingModule, RouterTestingModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LogsPageComponent);
    logService = TestBed.inject(LogService);
    timezoneService = TestBed.inject(TimezoneService);
    statusMessageService = TestBed.inject(StatusMessageService);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when page is still loading', () => {
    component.isLoading = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when searching for details in search form', () => {
    component.formModel = {
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
      filters: {
        severity: LogSeverity.ERROR,
        minSeverity: LogSeverity.INFO,
        logEvent: LogEvent.REQUEST_LOG,
      },
    };
    component.filterType = 'EVENT';
    component.isLoading = false;
    component.isSearching = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should search for logs when search button is clicked', () => {
    const logSpy: Spy = spyOn(logService, 'searchLogs').and
        .returnValue(of({ logEntries: [] }));
    const timeSpy: Spy = spyOn(timezoneService, 'resolveLocalDateTime').and
        .returnValue(0);

    component.isLoading = false;
    component.isSearching = false;
    component.filterType = 'SEVERITY';
    component.formModel = {
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
      filters: {
        severity: LogSeverity.INFO,
        minSeverity: LogSeverity.INFO,
        logEvent: LogEvent.REQUEST_LOG,
      },
    };
    fixture.detectChanges();

    fixture.debugElement.nativeElement.querySelector('#query-button').click();

    expect(timeSpy).toHaveBeenCalledTimes(2);
    expect(logSpy).toHaveBeenCalledWith({
      startTime: 0,
      endTime: 0,
      order: 'desc',
      severity: LogSeverity.INFO,
      minSeverity: undefined,
      logEvent: undefined,
    });
  });

  it('should search for logs with minimum severity', () => {
    const logSpy: Spy = spyOn(logService, 'searchLogs').and
        .returnValue(of({ logEntries: [] }));
    const timeSpy: Spy = spyOn(timezoneService, 'resolveLocalDateTime').and
        .returnValue(0);

    component.isLoading = false;
    component.isSearching = false;
    component.filterType = 'MIN_SEVERITY';
    component.formModel = {
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
      filters: {
        severity: LogSeverity.INFO,
        minSeverity: LogSeverity.INFO,
        logEvent: LogEvent.REQUEST_LOG,
      },
    };
    fixture.detectChanges();

    fixture.debugElement.nativeElement.querySelector('#query-button').click();

    expect(timeSpy).toHaveBeenCalledTimes(2);
    expect(logSpy).toHaveBeenCalledWith({
      startTime: 0,
      endTime: 0,
      order: 'desc',
      severity: undefined,
      minSeverity: LogSeverity.INFO,
      logEvent: undefined,
    });
  });

  it('should search for logs with event type', () => {
    const logSpy: Spy = spyOn(logService, 'searchLogs').and
        .returnValue(of({ logEntries: [] }));
    const timeSpy: Spy = spyOn(timezoneService, 'resolveLocalDateTime').and
        .returnValue(0);

    component.isLoading = false;
    component.isSearching = false;
    component.filterType = 'EVENT';
    component.formModel = {
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
      filters: {
        severity: LogSeverity.INFO,
        minSeverity: LogSeverity.INFO,
        logEvent: LogEvent.REQUEST_LOG,
        traceId: 'testTrace',
        sourceLocation: {
          file: 'testFile',
          line: 0,
          function: 'testFunction',
        },
        userInfoParams: {
          googleId: 'testGoogleId',
          regkey: '',
          email: '',
        },
      },
    };
    fixture.detectChanges();

    fixture.debugElement.nativeElement.querySelector('#query-button').click();

    expect(timeSpy).toHaveBeenCalledTimes(2);
    expect(logSpy).toHaveBeenCalledWith({
      startTime: 0,
      endTime: 0,
      order: 'desc',
      severity: undefined,
      minSeverity: undefined,
      logEvent: LogEvent.REQUEST_LOG,
      traceId: 'testTrace',
      sourceLocation: {
        file: 'testFile',
        line: 0,
        function: 'testFunction',
      },
      userInfoParams: {
        googleId: 'testGoogleId',
        regkey: '',
        email: '',
      },
    });
  });

  it('should display error message if source function is filled and source file is empty', () => {
    component.isLoading = false;
    component.isSearching = false;
    component.filterType = 'SEVERITY';
    component.formModel = {
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
      filters: {
        severity: LogSeverity.INFO,
        minSeverity: LogSeverity.INFO,
        logEvent: LogEvent.REQUEST_LOG,
        sourceLocation: {
          file: '',
          line: 0,
          function: 'testFunction',
        },
      },
    };
    const spy: Spy = spyOn(statusMessageService, 'showErrorToast');
    fixture.detectChanges();
    fixture.debugElement.nativeElement.querySelector('#query-button').click();
    expect(spy).lastCalledWith('Please fill in Source location file or clear Source location function');
  });

  it('should disable load button if there is no next page', () => {
    spyOn(logService, 'searchLogs').and.returnValue(of({ logEntries: [] }));
    spyOn(timezoneService, 'resolveLocalDateTime').and.returnValue(0);
    component.formModel = {
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
      filters: {
        severity: LogSeverity.INFO,
        minSeverity: LogSeverity.INFO,
        logEvent: LogEvent.REQUEST_LOG,
      },
    };
    component.filterType = 'SEVERITY';
    component.isSearching = false;
    component.hasResult = true;
    component.searchForLogs();
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#load-previous-button');
    expect(button).toBeNull();
  });

  it('should search for all error logs when search button is clicked', () => {
    const testLog1: GeneralLogEntry = {
      severity: LogSeverity.ERROR,
      trace: 'testTrace1',
      insertId: 'testInsertId1',
      resourceIdentifier: {},
      sourceLocation: {
        file: 'file1',
        line: 10,
        function: 'function1',
      },
      timestamp: 1549095330000,
      message: 'message',
    };
    const logSpy: Spy = spyOn(logService, 'searchLogs').and
      .returnValues(of({ logEntries: [testLog1], hasNextPage: true }), of({ logEntries: [], hasNextPage: false }));
    const timeSpy: Spy = spyOn(timezoneService, 'resolveLocalDateTime').and
      .returnValue(0);

    component.isLoading = false;
    component.isSearching = false;
    component.isTableView = false;

    fixture.detectChanges();
    fixture.debugElement.nativeElement.querySelector('#query-button').click();

    expect(timeSpy).toHaveBeenCalledTimes(2);
    expect(logSpy).toHaveBeenCalledTimes(2);
    expect(logSpy.calls.mostRecent().args).toEqual([{
      startTime: 0,
      endTime: 1549095330000,
      order: 'desc',
      severity: 'ERROR',
    }]);
  });

  it('should sort logs based on source location', () => {
    component.isTableView = false;
    const testLog1: GeneralLogEntry = {
      severity: LogSeverity.ERROR,
      trace: 'testTrace1',
      insertId: 'testInsertId1',
      resourceIdentifier: {},
      sourceLocation: {
        file: 'file1',
        line: 10,
        function: 'function1',
      },
      timestamp: 1549095330000,
      message: 'message',
    };
    const testLog2: GeneralLogEntry = {
      severity: LogSeverity.ERROR,
      trace: 'testTrace2',
      insertId: 'testInsertId2',
      resourceIdentifier: {},
      sourceLocation: {
        file: 'file2',
        line: 10,
        function: 'function1',
      },
      timestamp: 1549095330000,
      message: 'message',
    };
    const testLog3: GeneralLogEntry = {
      severity: LogSeverity.ERROR,
      trace: 'testTrace3',
      insertId: 'testInsertId3',
      resourceIdentifier: {},
      sourceLocation: {
        file: 'file2',
        line: 10,
        function: 'function1',
      },
      timestamp: 1549095330000,
      message: 'message',
    };
    spyOn(logService, 'searchLogs').and
      .returnValue(of({ logEntries: [testLog1, testLog2, testLog3] }));
    spyOn(timezoneService, 'resolveLocalDateTime').and.returnValue(0);

    component.isLoading = false;
    component.isSearching = false;
    component.isTableView = false;
    component.searchForLogs();

    expect(component.histogramResult.length).toEqual(2);
  });
});
