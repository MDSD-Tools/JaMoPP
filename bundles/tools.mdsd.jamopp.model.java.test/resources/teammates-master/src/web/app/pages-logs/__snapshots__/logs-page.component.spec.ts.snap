// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogsPageComponent should snap when page is still loading 1`] = `
<tm-logs-page
  ACTION_CLASSES={[Function Array]}
  EVENTS={[Function Array]}
  LOGS_RETENTION_PERIOD_IN_DAYS={[Function Number]}
  LOGS_RETENTION_PERIOD_IN_MILLISECONDS={[Function Number]}
  SEVERITIES={[Function Array]}
  activatedRoute={[Function ActivatedRoute]}
  dateToday={[Function Object]}
  earliestLogTimestampRetrieved={[Function Number]}
  earliestSearchDate={[Function Object]}
  filterType={[Function String]}
  formModel={[Function Object]}
  hasNextPage="false"
  hasPreviousPage={[Function Boolean]}
  hasResult="false"
  histogramResult={[Function Array]}
  isAdmin="undefined"
  isFiltersExpanded="false"
  isLoading={[Function Boolean]}
  isSearching="false"
  isTableView={[Function Boolean]}
  latestLogTimestampRetrieved="0"
  logService={[Function LogService]}
  logsMap={[Function Map]}
  queryParams={[Function Object]}
  searchEndTime="0"
  searchResults={[Function Array]}
  searchStartTime="0"
  statusMessageService={[Function StatusMessageService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Logs
  </h1><p>
    This page allows you to view the logs in the past 30 days.
  </p><button
    class="btn btn-info"
  >
    Switch to histogram view
  </button><hr /><tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner><div>
    
    
    <div
      class="center mb-4"
    >
      
    </div>
    
    
    <div
      class="center mb-4"
    >
      
    </div>
    
  </div>
</tm-logs-page>
`;

exports[`LogsPageComponent should snap when searching for details in search form 1`] = `
<tm-logs-page
  ACTION_CLASSES={[Function Array]}
  EVENTS={[Function Array]}
  LOGS_RETENTION_PERIOD_IN_DAYS={[Function Number]}
  LOGS_RETENTION_PERIOD_IN_MILLISECONDS={[Function Number]}
  SEVERITIES={[Function Array]}
  activatedRoute={[Function ActivatedRoute]}
  dateToday={[Function Object]}
  earliestLogTimestampRetrieved={[Function Number]}
  earliestSearchDate={[Function Object]}
  filterType={[Function String]}
  formModel={[Function Object]}
  hasNextPage="false"
  hasPreviousPage={[Function Boolean]}
  hasResult="false"
  histogramResult={[Function Array]}
  isAdmin="undefined"
  isFiltersExpanded="false"
  isLoading="false"
  isSearching={[Function Boolean]}
  isTableView={[Function Boolean]}
  latestLogTimestampRetrieved="0"
  logService={[Function LogService]}
  logsMap={[Function Map]}
  queryParams={[Function Object]}
  searchEndTime="0"
  searchResults={[Function Array]}
  searchStartTime="0"
  statusMessageService={[Function StatusMessageService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Logs
  </h1><p>
    This page allows you to view the logs in the past 30 days.
  </p><button
    class="btn btn-info"
  >
    Switch to histogram view
  </button><hr /><div
    class="card bg-form no-border"
  >
    <div
      class="card-body"
    >
      <div
        class="row form-group"
      >
        <div
          class="col-6"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-from-datepicker"
            >
              Search period from
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-from-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
              id="logs-from-timepicker"
            >
              <ngb-timepicker
                class="ng-untouched ng-pristine ng-valid"
              >
                <fieldset>
                  <div
                    class="ngb-tp"
                  >
                    <div
                      class="ngb-tp-input-container ngb-tp-hour"
                    >
                      
                      <input
                        aria-label="Hours"
                        class="ngb-tp-input form-control"
                        inputmode="numeric"
                        maxlength="2"
                        placeholder="HH"
                        type="text"
                      />
                      
                    </div>
                    <div
                      class="ngb-tp-spacer"
                    >
                      :
                    </div>
                    <div
                      class="ngb-tp-input-container ngb-tp-minute"
                    >
                      
                      <input
                        aria-label="Minutes"
                        class="ngb-tp-input form-control"
                        inputmode="numeric"
                        maxlength="2"
                        placeholder="MM"
                        type="text"
                      />
                      
                    </div>
                    
                    
                    
                    
                  </div>
                </fieldset>
              </ngb-timepicker>
            </div>
          </div>
          <div
            class="col-12 mb-1 mt-4"
          >
            <label
              class="font-weight-bold"
              for="logs-to-datepicker"
            >
              Search period until
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-to-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
              id="logs-to-timepicker"
            >
              <ngb-timepicker
                class="ng-untouched ng-pristine ng-valid"
              >
                <fieldset>
                  <div
                    class="ngb-tp"
                  >
                    <div
                      class="ngb-tp-input-container ngb-tp-hour"
                    >
                      
                      <input
                        aria-label="Hours"
                        class="ngb-tp-input form-control"
                        inputmode="numeric"
                        maxlength="2"
                        placeholder="HH"
                        type="text"
                      />
                      
                    </div>
                    <div
                      class="ngb-tp-spacer"
                    >
                      :
                    </div>
                    <div
                      class="ngb-tp-input-container ngb-tp-minute"
                    >
                      
                      <input
                        aria-label="Minutes"
                        class="ngb-tp-input form-control"
                        inputmode="numeric"
                        maxlength="2"
                        placeholder="MM"
                        type="text"
                      />
                      
                    </div>
                    
                    
                    
                    
                  </div>
                </fieldset>
              </ngb-timepicker>
            </div>
          </div>
        </div>
        
        <div
          class="col-4"
        >
          <div
            class="col-12"
          >
            <label
              class="font-weight-bold"
              for="log-severity-radio"
            >
              Log severity level / event
            </label>
          </div>
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
              id="log-severity-radio"
            >
              <div>
                <input
                  class="form-check-input mt-3 ng-untouched ng-pristine ng-valid"
                  id="severity"
                  name="severity"
                  type="radio"
                />
                <label
                  for="severity"
                >
                  severity =Â 
                </label>
                <select
                  class="form-control inline ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="INFO"
                  >
                    INFO
                  </option>
                  <option
                    value="WARNING"
                  >
                    WARNING
                  </option>
                  <option
                    value="ERROR"
                  >
                    ERROR
                  </option>
                </select>
              </div>
              <br />
              <div>
                <input
                  class="form-check-input mt-3 ng-untouched ng-pristine ng-valid"
                  id="min-severity"
                  name="severity"
                  type="radio"
                />
                <label
                  for="min-severity"
                >
                  severity &gt;=Â 
                </label>
                <select
                  class="form-control inline ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="INFO"
                  >
                    INFO
                  </option>
                  <option
                    value="WARNING"
                  >
                    WARNING
                  </option>
                  <option
                    value="ERROR"
                  >
                    ERROR
                  </option>
                </select>
              </div>
              <br />
              <div>
                <input
                  class="form-check-input mt-3 ng-untouched ng-pristine ng-valid"
                  id="event"
                  name="severity"
                  type="radio"
                />
                <label
                  for="event"
                >
                  All event logs Â 
                </label>
                <select
                  class="form-control inline ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="REQUEST_LOG"
                  >
                    REQUEST_LOG
                  </option>
                  <option
                    value="EXCEPTION_LOG"
                  >
                    EXCEPTION_LOG
                  </option>
                  <option
                    value="EMAIL_SENT"
                  >
                    EMAIL_SENT
                  </option>
                  <option
                    value="FEEDBACK_SESSION_AUDIT"
                  >
                    FEEDBACK_SESSION_AUDIT
                  </option>
                  <option
                    value="INSTANCE_LOG"
                  >
                    INSTANCE_LOG
                  </option>
                </select>
              </div>
              <br />
            </label>
          </div>
        </div>
        <div
          class="col-1 mt-4"
        >
          <button
            class="float-right btn btn-primary"
            id="query-button"
          >
            Query
          </button>
        </div>
      </div>
      
      <div
        class="row form-group"
      >
        <div
          class="col-12"
        >
          <div
            class="col-12 mb-1 mt-2"
          >
            <button
              class="btn btn-success"
            >
              <i
                class="fas fa-plus"
              />
               Advanced filters 
            </button>
          </div>
        </div>
      </div>
      
    </div>
  </div><div>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
  </div>
</tm-logs-page>
`;

exports[`LogsPageComponent should snap with default fields 1`] = `
<tm-logs-page
  ACTION_CLASSES={[Function Array]}
  EVENTS={[Function Array]}
  LOGS_RETENTION_PERIOD_IN_DAYS={[Function Number]}
  LOGS_RETENTION_PERIOD_IN_MILLISECONDS={[Function Number]}
  SEVERITIES={[Function Array]}
  activatedRoute={[Function ActivatedRoute]}
  dateToday={[Function Object]}
  earliestLogTimestampRetrieved={[Function Number]}
  earliestSearchDate={[Function Object]}
  filterType={[Function String]}
  formModel={[Function Object]}
  hasNextPage="false"
  hasPreviousPage={[Function Boolean]}
  hasResult="false"
  histogramResult={[Function Array]}
  isAdmin="undefined"
  isFiltersExpanded="false"
  isLoading={[Function Boolean]}
  isSearching="false"
  isTableView={[Function Boolean]}
  latestLogTimestampRetrieved="0"
  logService={[Function LogService]}
  logsMap={[Function Map]}
  queryParams={[Function Object]}
  searchEndTime="0"
  searchResults={[Function Array]}
  searchStartTime="0"
  statusMessageService={[Function StatusMessageService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Logs
  </h1><p>
    This page allows you to view the logs in the past 30 days.
  </p><button
    class="btn btn-info"
  >
    Switch to histogram view
  </button><hr /><tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner><div>
    
    
    <div
      class="center mb-4"
    >
      
    </div>
    
    
    <div
      class="center mb-4"
    >
      
    </div>
    
  </div>
</tm-logs-page>
`;
